packages/next/src/build/swc/loaderWorkerPool.ts

```diff
@@ -0,0 +1,42 @@
+import { Worker } from 'worker_threads'
+import type {
+  WorkerCreationParams,
+  WorkerTermination,
+} from './generated-native'
+
+const loaderWorkers: Record<string, Array<Worker>> = {}
+
+export async function runLoaderWorkerPool(
+  binding: typeof import('./generated-native'),
+  bindingPath: string
+) {
+  binding.registerWorkerCreator((request: WorkerCreationParams) => {
+    const { options, taskId } = request
+    const { filename, cwd } = options
+
+    const worker = new Worker(filename, {
+      workerData: {
+        poolId: filename,
+        bindingPath,
+        cwd,
+      },
+    })
+
+    const workers = loaderWorkers[filename] || (loaderWorkers[filename] = [])
```

Comment:
```suggestion
    const workers = loaderWorkers[poolId] || (loaderWorkers[poolId] = [])
```

Sounds like cwd should also part of the key.